apps/system/js/multi_screen_controller.js

```diff
@@ -211,25 +158,34 @@
         }));
       });
     },
-    _displayListResultHandler: function(detail) {
-      if (!this.queryPromiseCallback) {
-        return;
-      }
 
-      switch (detail.type) {
-        case 'get-display-list-success':
-          this.queryPromiseCallback.resolve(detail.display);
-          break;
-        case 'get-display-list-error':
-          this.queryPromiseCallback.reject(detail.error);
+    _handle_mozPresentationChromeEvent: function(evt) {
+      this.debug('got mozPresentationChromeEvent event');
+
+      var detail = evt.detail;
+      switch(detail.type) {
+        case 'presentation-launch-receiver':
+          // target screen ID
+          this.screenId = this.requestDeviceId.split(':')[1];
```

Comment:
How about using`this.requestDeviceId` directly? Otherwise it's hard to understand what the `screenId` stands for.